File: ruby/test/wire/context_registry_test.rb
# frozen_string_literal: true

require_relative "wiremock_test_case"

class ContextRegistryWireTest < WireMockTestCase
  def setup
    super

    @client = Sikaru::Client.new(
      api_key: "<token>",
      base_url: WIREMOCK_BASE_URL
    )
  end

  def test_context_registry_create_context_registry_change_with_wiremock
    test_id = "context_registry.create_context_registry_change.0"

    @client.context_registry.create_context_registry_change(
      project_id: "project_id",
      name: "name",
      repo_id: "repoId",
      request_options: {
        additional_headers: {
          "X-Test-Id" => "context_registry.create_context_registry_change.0"
        }
      }
    )

    verify_request_count(
      test_id: test_id,
      method: "POST",
      url_path: "/v1/projects/project_id/context-registry",
      query_params: nil,
      expected: 1
    )
  end
end

File: rust/tests/context_registry_test.rs
use sikaru::prelude::*;

mod wire_test_utils;

#[tokio::test]
#[allow(unused_variables, unreachable_code)]
async fn test_context_registry_create_context_registry_change_with_wiremock() {
    wire_test_utils::reset_wiremock_requests().await.unwrap();
    let wiremock_base_url = wire_test_utils::get_wiremock_base_url();

    let mut config = ClientConfig {
        token: Some("<token>".to_string()),
        ..Default::default()
    };
    config.base_url = wiremock_base_url.to_string();
    let client = Sikaru::new(config).expect("Failed to build client");

    let result = client
        .context_registry
        .create_context_registry_change(
            &"project_id".to_string(),
            &CreateContextRegistryChangeRequest {
                name: "name".to_string(),
                repo_id: "repoId".to_string(),
                action: None,
                files: None,
                kind: None,
                linked_signal: None,
                owner: None,
                production_commit: None,
                production_tag: None,
                repo_type: None,
                source_trace_ids: None,
                staging_commit: None,
                staging_tag: None,
                tools: None,
                variables: None,
                visibility: None,
            },
            None,
        )
        .await;

    assert!(result.is_ok(), "Client method call should succeed");

    wire_test_utils::verify_request_count(
        "POST",
        "/v1/projects/project_id/context-registry",
        None,
        1,
    )
    .await
    .unwrap();
}
